### PR TITLE
feat: Department CRUD, students of course and courses of student.

### PR DESCRIPTION
**Pull request by bynalab**

### CODE REVIEW

File: app/Http/Controllers/Api/CourseController.php [hunk 1]
```diff
@@ -336,6 +335,46 @@ public function getStudentEnrolledCourses()
...
+        if ($validator->fails()) {
+            return response()->json(['error' => "course_id field is required"], 400);
+        }
+
+        $courseId = $request->get('course_id');
+        $courseStudent = Course::where('id',$courseId)->with('students')->withCount('students')->get();
+        if($courseStudent){
+            return response()->json($courseStudent, 200);
+        }
+        return response()->json(['success' => false], 400);
```
Comment (Auwalms) [inline app/Http/Controllers/Api/CourseController.php:358]: Why is success having a false returned and 400 status 😩😩😩😩.  Is PHP confusing me ?

**Review by Auwalms — COMMENTED** [review]
Can you please explain what happens in the comment I left? Wondering why 400 and success are in the same line @bynalab

File: app/Http/Controllers/Api/CourseController.php (hunk 1 above)
Comment (bynalab) [reply app/Http/Controllers/Api/CourseController.php:358]: Lol. Okay, so success is just a key right, and it's value is a Boolean, say can be successful or not.
Though, if you think there is a better approach to this, of cause we can implement that.
This is an unknown error, so I just throw a bad request error(400).

Comment (Auwalms) [reply app/Http/Controllers/Api/CourseController.php:358]: Oh cool, the explanation makes sense. we go this route.

**Review by Auwalms — APPROVED** [review]
GTG

**Review by abdullahikawu — APPROVED** [review]
Looks good.

**Review by Auwalms — APPROVED** [review]

**Pull request merged**